config/crd/bases/kuadrant.io_dnspolicies.yaml

```diff
@@ -0,0 +1,156 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.10.0
+  creationTimestamp: null
+  name: dnspolicies.kuadrant.io
+spec:
+  group: kuadrant.io
+  names:
+    kind: DNSPolicy
+    listKind: DNSPolicyList
+    plural: dnspolicies
+    singular: dnspolicy
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - description: DNSPolicy ready.
+      jsonPath: .status.conditions[?(@.type=="Ready")].status
+      name: Ready
+      type: string
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: DNSPolicy is the Schema for the dnspolicies API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: DNSPolicySpec defines the desired state of DNSPolicy
+            properties:
+              dnsRecordRef:
+                properties:
+                  name:
+                    type: string
+                  namespace:
+                    type: string
+                required:
+                - name
+                - namespace
+                type: object
+              healthCheck:
```

Comment:
good to add some comment here: # by default this health check will be applied to each unique DNS A Record for the listeners assigned to the target gateway